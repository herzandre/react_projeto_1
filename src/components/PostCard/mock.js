export const postCardPropsMock = {
    title: 'title-test',
    body: 'body-test',
    id: 1,
    cover: 'img/img.png'
}