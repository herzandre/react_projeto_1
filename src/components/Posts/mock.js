export const postsPropsMock =  {
    posts: [
        {
            id: 1,
            title: 'title-test 1',
            body: 'body-test 1',
            cover: 'img/img1.jpg'
        },
        {
            id: 2,
            title: 'title-test 2',
            body: 'body-test 2',
            cover: 'img/img2.jpg'
        },
        {
            id: 3,
            title: 'title-test 3',
            body: 'body-test 3',
            cover: 'img/img3.jpg'
        }
    ]
}