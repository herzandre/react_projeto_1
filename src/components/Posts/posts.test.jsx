import { render, screen } from '@testing-library/react';
import { Posts } from '.';
import { postsPropsMock } from './mock';

describe('<Posts />', () => {
  it('should render Posts correctly"', () => {
    render(<Posts {...postsPropsMock} />);

    expect(screen.getAllByRole('heading', { name: /title-test/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title-test/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);

    expect(screen.getByRole('img', { name: /title-test 3/i })).toHaveAttribute('src', 'img/img3.jpg');
    expect(screen.getByRole('img', { name: /title-test 2/i })).toHaveAttribute('src', 'img/img2.jpg');
    expect(screen.getByRole('img', { name: /title-test 2/i })).toHaveAttribute('src', 'img/img2.jpg');
  });

  it('should not render posts', () => {
    render(<Posts />);

    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...postsPropsMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
