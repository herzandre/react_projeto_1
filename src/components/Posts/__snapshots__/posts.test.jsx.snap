// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
    id="1"
  >
    <img
      alt="title-test 1"
      src="img/img1.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
         
        title-test 1
         
        1
         
      </h2>
      <p>
         
        body-test 1
         
      </p>
    </div>
  </div>
  <div
    class="post"
    id="2"
  >
    <img
      alt="title-test 2"
      src="img/img2.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
         
        title-test 2
         
        2
         
      </h2>
      <p>
         
        body-test 2
         
      </p>
    </div>
  </div>
  <div
    class="post"
    id="3"
  >
    <img
      alt="title-test 3"
      src="img/img3.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
         
        title-test 3
         
        3
         
      </h2>
      <p>
         
        body-test 3
         
      </p>
    </div>
  </div>
</div>
`;
